odoo/tools/mail.py

```diff
@@ -595,3 +595,30 @@ def formataddr(pair, charset='utf-8'):
                 name=email_addr_escapes_re.sub(r'\\\g<0>', name),
                 addr=address)
     return address
+
+
+def email_domain_extract(email):
+    """Return the domain of the given email."""
+    if not email:
+        return
+
+    email_split = getaddresses([email])
+    if not email_split or not email_split[0]:
+        return
+
+    *_, domain = email_split[0][1].rpartition('@')
+    return domain
+
+
+def encapsulate_email(old_email, new_email):
```

Comment:
We should probably be safe and take only email-part of new_email. If someone uses it and gives new_email as ``Robert <robert@example.com>`` you could have results like ``"Martine (martine@example.com)" <Robert <robert@example.com>>``

Reply:
indeed :)